src/polygon.jl

```diff
@@ -0,0 +1,69 @@
+"""
+    getlevel(dataset::ArchGDAL.AbstractDataset, reqlevel::Int) 
+Get layer of the desired level from the dataset
+"""
+function getlevel(dataset::ArchGDAL.AbstractDataset, reqlevel::Int)
+    # number of layers in the dataset
+    layers = ArchGDAL.nlayer(dataset)
+
+    for i = 0:layers-1
+        layer = ArchGDAL.getlayer(dataset, i)
+        layername = ArchGDAL.getname(layer)
+        layerlevel = split(layername, "_")[end]
+        if layerlevel == string(reqlevel)
+            return layer
+        end
+    end
+
+    throw(ArgumentError("valid levels for the given dataset are 0...$(layers-1)"))
+end
+
+"""
+    isgpkg(file)::Bool
+Returns true for .gpkg files
+"""
+function isgpkg(file)::Bool
```

Comment:
I would remove the ::Bool annotation. These are good assertions, but the code looks inconsistent with some function having annotations and some others without annotations.